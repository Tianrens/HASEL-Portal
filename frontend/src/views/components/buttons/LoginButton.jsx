import { Icon } from '@mui/material';
import React from 'react';
import { firebase, auth } from '../../../firebase';
import { StyledButton } from './StyledButton';

async function login(callback, errorCallback) {
    const provider = new firebase.auth.GoogleAuthProvider();

    await auth.signInWithPopup(provider).then(() => {
        const userEmail = auth.currentUser.email;
        if (userEmail.endsWith('@aucklanduni.ac.nz') || userEmail.endsWith('@auckland.ac.nz')) {
            callback();
        } else {
            auth.signOut();
            errorCallback('Not using a University of Auckland email.');
        }
    });
}

/**
 * Button to log in. Upon successful login, calls the callback and saves the authentication details in the AppProvider.
 * Upon unsuccessful login, throws error.
 * @param {function callback specifies the function that is called after the login is successful}
 */
function LoginButton({ callback, errorCallback }) {
    return (
        <StyledButton
            icon={<Icon>login</Icon>}
            onClick={() => {
                login(callback, errorCallback);
            }}
        >
            Login
        </StyledButton>
    );
}

export default LoginButton;
