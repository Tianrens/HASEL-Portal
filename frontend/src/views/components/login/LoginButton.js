import React from 'react';
import { Button } from '@material-ui/core';
import { firebase, auth } from '../../../firebase';

async function login(callback) {
    const provider = new firebase.auth.GoogleAuthProvider();

    await auth.signInWithPopup(provider).then(() => {

        const userEmail = auth.currentUser.email;
        if (userEmail.endsWith('@aucklanduni.ac.nz') || userEmail.endsWith('@auckland.ac.nz')) {
            callback();
        } else {
            auth.signOut();
            throw new Error('not using an University of Auckland email');
        }

    });
}

/**
 * Button to log in. Upon successful login, calls the callback and saves the authentication details in the AppProvider.
 * Upon unsuccessful login, throws error.
 * @param {function callback specifies the function that is called after the login is successful}
 */
export function LoginButton( {callback}) {
    return (
        <Button
            variant='contained'
            color='primary'
            onClick={() => {login(callback);}}
        >
            Log In
        </Button>
    );
}

