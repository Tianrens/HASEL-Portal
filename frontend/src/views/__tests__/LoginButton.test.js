import React from 'react';
import renderer from 'react-test-renderer';
import { StylesProvider } from '@material-ui/core';
import LoginButton from '../components/buttons/LoginButton';

test('Login Button renders properly', () => {
    const snapshotComponent = renderer.create(
        <StylesProvider injectFirst>
            <LoginButton />
        </StylesProvider>,
    );
    const tree = snapshotComponent.toJSON();
    expect(tree).toMatchSnapshot();
});
