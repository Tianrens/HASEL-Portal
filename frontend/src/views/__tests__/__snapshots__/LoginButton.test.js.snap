// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Button renders properly 1`] = `
<button
  className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButtonBase-root button undefined     css-jh47zj-MuiButtonBase-root-MuiButton-root"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-startIcon MuiButton-iconSizeLarge css-htszrh-MuiButton-startIcon"
  >
    <span
      aria-hidden={true}
      className="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-kp9ftd-MuiIcon-root"
    >
      login
    </span>
  </span>
  Login
</button>
`;
